openedx/core/lib/api/authentication.py

```diff
@@ -35,6 +35,18 @@ def authenticate(self, request):
         fails.
         """
         set_custom_metric("OAuth2AuthenticationDeprecated", "Failed")
+
+        auth = get_authorization_header(request).split()
+
+        if auth and len(auth) == 2 and auth[0].lower() == b'bearer':
+            access_token = auth[1].decode('utf8')
+            set_custom_metric('OAuth2Authentication_token_location', 'bearer-in-header')
+        elif 'access_token' in request.POST:
+            access_token = request.POST['access_token']
+            set_custom_metric('OAuth2Authentication_token_location', 'post-token')
+
+        set_custom_metric("OAuth2Authentication_token_parts", len(access_token.split('.')))
```

Comment:
Just to be safe, maybe put all of this in a try/catch to ensure we don't break anything by mistake.  In the catch, you are welcome to add another metric for `OAuth2Authentication_metric_error` with the exception message.